projects/ion/src/lib/step/step.component.ts

```diff
@@ -0,0 +1,139 @@
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  OnInit,
+  Output,
+  SimpleChange,
+  SimpleChanges,
+} from '@angular/core';
+
+export interface StepType {
+  label: string;
+  description?: string;
+  index?: number;
+  status?: 'default' | 'selected' | 'checked' | 'error';
+}
+
+export type StepTheme = 'default' | 'pastel-tones' | 'dark-tones';
+
+export type CrossType = 'initial' | 'final';
+
+@Component({
+  selector: 'ion-step',
+  templateUrl: './step.component.html',
+  styleUrls: ['./step.component.scss'],
+})
+export class StepComponent implements OnInit, OnChanges {
+  @Input() current = 1;
+  @Input() disabled = false;
+  @Input() theme: StepTheme = 'default';
+  @Input() steps: StepType[];
+  @Output() indexChange = new EventEmitter<number>();
+
+  public indexShowed = 1;
+
+  public startSteps(): void {
+    let index = 1;
+    const firstStep = this.steps[0];
+
+    firstStep.status = firstStep.status ? firstStep.status : 'selected';
+
+    this.steps.map((step, indexArray) => {
+      const previusStep = this.steps[indexArray - 1];
+      const actualStepStatus = step.status;
+      if (
+        actualStepStatus !== 'checked' &&
+        previusStep &&
+        previusStep.status === 'checked'
```

Comment:
```suggestion
        actualStepStatus !== 'checked' &&
        previusStep &&
        previusStep.status === 'checked'
```

fix typo: previous. Create a function to return this boolean.